src/agentscope/models/dashscope_model.py

```diff
@@ -0,0 +1,431 @@
+             **kwargs (`Any`):
+                 The keyword arguments to DashScope Image Synthesis API,
+                 e.g. `n`, `size`, etc. Please refer to
+                 https://help.aliyun.com/zh/dashscope/developer-reference/api-details-9
+        for more detailed arguments.
+
+         Returns:
+             `ModelResponse`:
+                 A list of image urls in image_urls field and the
+                 raw response in raw field.
+
+         Note:
+             `parse_func`, `fault_handler` and `max_retries` are reserved
+             for `_response_parse_decorator` to parse and check the
+             response generated by model wrapper. Their usages are listed
+             as follows:
+                 - `parse_func` is a callable function used to parse and
+                 check the response generated by the model, which takes
+                 the response as input.
+                 - `max_retries` is the maximum number of retries when the
+                 `parse_func` raise an exception.
+                 - `fault_handler` is a callable function which is called
+                 when the response generated by the model is invalid after
+                 `max_retries` retries.
+        """
+        # step1: prepare keyword arguments
+        kwargs = {**self.generate_args, **kwargs}
+
+        # step2: forward to generate response
+        response = dashscope.ImageSynthesis.call(
+            model=self.model,
+            prompt=prompt,
+            n=1,
+            **kwargs,
+        )
+        if response.status_code != HTTPStatus.OK:
+            error_msg = (
+                f" Request id: {response.request_id},"
+                f" Status code: {response.status_code},"
+                f" error code: {response.code},"
+                f" error message: {response.message}."
+            )
+            raise RuntimeError(error_msg)
+
+        # step3: record the model api invocation if needed
+        self._save_model_invocation(
+            arguments={
+                "model": self.model,
+                "prompt": prompt,
+                **kwargs,
+            },
+            json_response=response,
+        )
+
+        # step4: update monitor accordingly
+        try:
+            self.monitor.update(
+                response.usage,
+                prefix=self.model,
+            )
+        except Exception as e:
+            logger.error(e)
+
+        # step5: return response
+        images = response["output"]["results"]
+        # Get image urls as a list
+        urls = [_["url"] for _ in images]
+
+        if save_local:
+            # Return local url if save_local is True
+            urls = [file_manager.save_image(_) for _ in urls]
+        return ModelResponse(image_urls=urls, raw=response)
+
+
+class DashScopeTextEmbeddingWrapper(DashScopeWrapper):
+    """The model wrapper for DashScope Text Embedding API."""
+
+    model_type: str = "dashscope_textembedding"
```

Comment:
maybe unified into `dashscope_embedding`

Reply:
since dashscope has TextEmbedding and MultiModalEmbedding, I change it to dashscope_text_embedding